crates/bevy_render/src/mesh/mesh.rs

```diff
@@ -294,18 +294,84 @@ impl Mesh {
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute(&self, name: impl Into<Cow<'static, str>>) -> Option<&VertexAttributeValues> {
         self.attributes.get(&name.into())
     }
 
+    /// Retrieves a mutable reference to the data currently set behind the given vertex attribute.
+    /// Returns None if the mesh has no data for this vertex attribute.
     pub fn attribute_mut(
         &mut self,
         name: impl Into<Cow<'static, str>>,
     ) -> Option<&mut VertexAttributeValues> {
         self.attributes.get_mut(&name.into())
     }
 
+    /// Retrieves multiple mutable references to _distinct_ vertex attributes.
+    /// Returns None at index i if the mesh had no data for the vertex attribute given at index i.
+    ///
+    /// # Panics
+    /// Panics if any of the given attribute names resolve to the same attribute, as this would otherwise allow aliased mutability.
+    ///
+    /// # Example usage
+    /// ```
+    /// use crate::bevy_render::mesh::shape;
+    /// use crate::bevy_render::mesh::Mesh;
+    /// let mut mesh = Mesh::from(shape::Box::default());
+    /// let [positions, normals] = mesh.attribute_multi_mut([Mesh::ATTRIBUTE_POSITION, Mesh::ATTRIBUTE_NORMAL]);
+    /// ```
+    pub fn attribute_multi_mut<const N: usize>(
+        &mut self,
+        names: [impl Into<Cow<'static, str>>; N],
+    ) -> [Option<&mut VertexAttributeValues>; N] {
+        // SAFE: We verify that all the references returned are unique,
+        // therefore no aliased mutability is possible.
+        unsafe {
+            const INIT_POINTER: Option<*mut VertexAttributeValues> = None;
+            let mut pointers = [INIT_POINTER; N];
+
+            for (pointer, name) in pointers.iter_mut().zip(names) {
+                // Get raw mutable pointers to the attributes.
+                // Note that these pointers may indeed alias if the user gave duplicate names!
+                *pointer = self.attributes.get_mut(&name.into()).map(|a| a as *mut _);
+            }
+
+            // Verify that there are no duplicate pointers.
+            // This is critical to ensure no aliased mutability!
+            for i in 1..pointers.len() {
+                let current = match pointers[i - 1] {
+                    // No pointer, no chance for aliasing, so just continue to the next one.
+                    None => continue,
+                    Some(r) => r,
+                };
+                for maybe_pointer in pointers[i..].iter() {
+                    let other = match maybe_pointer {
+                        // No pointer, no chance for aliasing, so just continue to the next one.
+                        None => continue,
+                        Some(r) => r,
+                    };
+
+                    assert_ne!(
+                        current, *other,
+                        "Duplicate names given to attribute_multi_mut."
```

Comment:
It sounds like the problem isn't duplicate *names*, it's that the names provided point to the same attributes. Otherwise we could just check this at the top.

Reply:
I'm not sure I understand - how can the names be different but point to the same attribute?

Reply:
My bad, I wasn't sure if there was the possibility of synonymous names in a rendering context.

Why not just check string equality immediately?

Reply:
I'd have to gather up all the strings (calling .into()) ahead of time and then compare. I suppose it could work just as well but comparing the pointers seemed to more explicitly state the requirement of no aliased mutability.

Also its just faster :)

Reply:
Perfect, I'm happy with this.